lib/codecs/tests/avro.rs

```diff
@@ -0,0 +1,88 @@
+use bytes::{Bytes, BytesMut};
+use codecs::{
+    decoding::format::AvroDeserializerConfig, decoding::format::Deserializer,
+    encoding::format::AvroSerializerConfig,
+};
+use rstest::*;
+use similar_asserts::assert_eq;
+use std::{
+    fs::File,
+    io::Read,
+    path::{Path, PathBuf},
+    str::from_utf8,
+};
+use tokio_util::codec::Encoder;
+use vector_core::{config::LogNamespace, event::Event};
+
+#[rstest]
+fn roundtrip_avro_fixtures2(
+    #[files("tests/data/avro/avro/*.avro")]
+    #[exclude(
+        ".*(date|decimal_var|duration|fixed|map|time_millis|timestamp_micros|timestamp_millis).avro"
```

Comment:
Are these failing?

Reply:
Yes, these failed because the difference of inner type of int which avro using `i32` but vrlvalue using `i64`. And the values of `map` may be serialized in different order.

Reply:
Could we relax/specialize the tests in order to cover these cases?

Reply:
I didn't find a easy method, avro uses `i32` for some types which will throw errors when meeting `i64` which used by VrlValue Integer.

Reply:
| test case        | error message                        | avro inner type       | vrl inner type |
| ---------------- | ------------------------------------ | --------------------- | -------------- |
| date             | Date expected, got Long              | Date(i32)             | i64            |
| decimal_var      | Decimal expected, got Array          | Decimal(Decimal)      | Vec<u8>        |
| duration         | Duration expected, got Array         | Duration(Duration)    | [u8; 12]       |
| fixed            | String expected for fixed, got Array | Fixed(usize, Vec<u8>) | Vec<u8>        |
| time_millis      | TimeMillis expected, got Long        | TimeMillis(i32)       | i64            |
| timestamp_micros | TimestampMicros expected, got String | TimestampMicros(i64)  | DateTime<Utc>  |
| timestamp_millis | TimestampMillis expected, got String | TimestampMillis(i64)  | DateTime<Utc>  |

Reply:
For `map` type, could I convert the two items test case to one item test case to relax the test?

Reply:
> For map type, could I convert the two items test case to one item test case to relax the test?

Yes, sounds good.

Reply:
I also left a comment on how to fix `AvroValue::Fixed(_, bytes)`. 

For all the other types we can [do this](https://github.com/vectordotdev/vector/pull/18636#discussion_r1362513762). Workarounds should be doable but it's sufficient to return an error and unblock this PR.